pkg/cloudprovider/providers/azure/azure_util.go

```diff
@@ -116,6 +125,204 @@ func (az *Cloud) getSecurityRuleID(securityRuleName string) string {
 		securityRuleName)
 }
 
+// returns the full identifier of a publicIPAddress.
+func (az *Cloud) getpublicIPAddressID(pipName string) string {
+	return fmt.Sprintf(
+		publicIPAddressIDTemplate,
+		az.SubscriptionID,
+		az.ResourceGroup,
+		pipName)
+}
+
+// select load balancer for the service in the cluster
+// the selection algorithm selectes the the load balancer with currently has
+// the minimum lb rules, there there are multiple LB's with same number of rules
+// it selects the first one (sorted based on name)
+func (az *Cloud) selectLoadBalancer(clusterName string, service *v1.Service, existingLBs *[]network.LoadBalancer, nodes []*v1.Node) (selectedLB *network.LoadBalancer, existsLb bool, err error) {
+	isInternal := requiresInternalLoadBalancer(service)
+	serviceName := getServiceName(service)
+	glog.V(3).Infof("selectLoadBalancer(%s): isInternal(%s) - start", serviceName, isInternal)
+	availabilitySetNames, err := az.getLoadBalancerAvailabilitySetNames(service, nodes)
+	if err != nil {
+		glog.Errorf("az.selectLoadBalancer: cluster (%s) service(%s) - az.getLoadBalancerAvailabilitySetNames failed, err=(%v)", clusterName, serviceName, err)
+		return nil, false, err
+	}
+	glog.Infof("selectLoadBalancer(%s): isInternal(%s) - availabilitysetsname %v", serviceName, isInternal, *availabilitySetNames)
+	mapExistingLBs := map[string]*network.LoadBalancer{}
+	for lbx := range *existingLBs {
+		lb := (*existingLBs)[lbx]
+		mapExistingLBs[*lb.Name] = &lb
+	}
+	selectedLBRuleCount := math.MaxInt32
+	for asx := range *availabilitySetNames {
+		currASName := (*availabilitySetNames)[asx]
+		currLBName := az.getLoadBalancerName(clusterName, currASName, isInternal)
+		lb, ok := mapExistingLBs[currLBName]
+		if !ok {
```

Comment:
I don't know what `ok` means here.  Is it more 'exists'?

Reply:
renamed to exists